Gcpe.Hub.API.IntegrationTests/SecuredPagesShould.cs

```diff
@@ -1,40 +1,54 @@
 ﻿using System.Net;
+using System.Net.Http;
 using System.Threading.Tasks;
 using FluentAssertions;
+using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
 using Xunit;
 
 namespace Gcpe.Hub.API.IntegrationTests
 {
     public class SecuredPagesShould : BaseWebApiTest
-    {
+    {
+
         public SecuredPagesShould(CustomWebApplicationFactory<Startup> factory) : base(factory)
         {
         }
 
-        [Theory(Skip = "Not available yet...")]
-        [InlineData("/SecurePage")]
-        public async Task RequireAnAuthenticatedUser(string url)
-        {
-            //-----------------------------------------------------------------------------------------------------------
-            // Arrange
-            //-----------------------------------------------------------------------------------------------------------
-            var client = _factory.CreateClient(
-                new WebApplicationFactoryClientOptions
-                {
-                    AllowAutoRedirect = false
-                });
+        [Theory]
+        [InlineData("/api/messages", "POST")]
+        [InlineData("/api/messages/123", "PUT")]
+        [InlineData("/api/messages/123", "DELETE")]
+        [InlineData("/api/posts", "POST")]
+        [InlineData("/api/posts/123", "PUT")]
+        [InlineData("/api/activities", "POST")]
+        [InlineData("/api/activities/123", "PUT")]
+        [InlineData("/api/socialmediaposts", "POST")]
+        [InlineData("/api/socialmediaposts/123", "PUT")]
+        [InlineData("/api/socialmediaposts/123", "DELETE")]
+        public async Task RequireAnAuthenticatedUser(string url, string requestType)
```

Comment:
Maybe add some tests for DoesNotRequireAnAuthenticatedUser

Reply:
done and reorganized the test setup a bit to make it a bit cleaner